cpp/autosar/test/rules/A7-1-2/test.cpp

```diff
@@ -204,4 +204,44 @@ class ExcludedCases {
 
   void operator=(ExcludedCases &) {}  // COMPLIANT
   void operator=(ExcludedCases &&) {} // COMPLIANT
-};
\ No newline at end of file
+};
+
+extern int random();
+constexpr int add(int x, int y) { return x + y; }
+constexpr int add1(int x, int y = 1) { return x + y; }
+constexpr int add2(int x, int y = add(add1(1), 2)) { return x + y; }
+constexpr int add3(int x, int y = random()) { return x + y; }
+constexpr int add4(int x = 1, int y = 2) { return x + y; }
+
+constexpr void fp_reported_in_466(int p) {
+  int x = add(1, 2); // NON_COMPLIANT
+  int y = add(1, p); // COMPLIANT
+
+  int z = 0;
+  if (p > 0) {
+    z = 1;
+  } else {
+    z = p;
+  }
+
+  int u = add(z, 2);  // COMPLIANT
+  int v = add(x, 2);  // NON_COMPLIANT
+  int w = add1(x, 2); // NON_COMPLIANT
+  int a = add1(x);    // NON_COMPLIANT
+  int b = add1(1);    // NON_COMPLIANT
+  int c = add1(1, z); // COMPLIANT
+  int d = add1(1, z); // COMPLIANT
+  int e = add1(z);    // COMPLIANT
+  int f = add2(1);    // NON_COMPLIANT
+  int g = add2(1, 2); // NON_COMPLIANT
+  int h = add2(x, 2); // NON_COMPLIANT
+  int i = add2(x, 2); // NON_COMPLIANT
```

Comment:
so I'll confess, I dont know the exact details of the constant propagation type logic that most compilers will use to checkup on if this can be `constexpr`... so I just added that specifier to all of the NON_COMPLIANT cases and the ones that disallow from the example compiling are : line 237,238,230,228 (maybe a few others)

additionally the readability of this section would benefit from some overhead comments that describe what each grouping exercises

Reply:
I've added clarifications to what is supposed to be tested. The compiler errors are caused because the functions receive a compile time constant but the variable couldn't be marked `constexpr` because the intermediate variable holding the compile time constant wasn't a `constexpr`. 

Because of the resulting compilation errors I`ve opted to not present all the result to prevent confusion. It does mean that a change can result in multiple new results, but that is the trade off that has to be made.